crypto/fipsmodule/sha/sha3_test.cc

```diff
@@ -496,6 +501,62 @@ TEST(KeccakInternalTest, SqueezeOutputBufferOverflow) {
   EVP_MD_unstable_sha3_enable(false);
 }
 
+// Test x4 batched SHAKE against 4 consecutive SHAKE calls
+// Assert success when digest and digest_x4 values are equal
+TEST(SHAKETest_x4, RandomMessages) {
+  KECCAK1600_CTX_x4 ctx;
+
+  uint8_t random_in[BATCHED_x4][RAND_BYTES_x4];
+  uint8_t digest[BATCHED_x4][RAND_OUT_BYTES_BLCKS * SHAKE128_BLOCKSIZE];
+  uint8_t digest_x4[BATCHED_x4][RAND_OUT_BYTES_BLCKS * SHAKE128_BLOCKSIZE];
+
+  // Test |SHAKE128_Init_x4|, |SHAKE128_Absorb_once_x4|, and |SHAKE128_Squeezeblocks_x4| functions
+  // Assert success when digest and digest_x4 values are equal
+  for (int i = 0; i < NUM_TESTS; i++) {
+    for (int j = 0; j < BATCHED_x4; j++) {
+      OPENSSL_memset(digest[j], 0, RAND_OUT_BYTES_BLCKS * SHAKE128_BLOCKSIZE);
+      OPENSSL_memset(digest_x4[j], 0, RAND_OUT_BYTES_BLCKS * SHAKE128_BLOCKSIZE);
+
+      ASSERT_TRUE(RAND_bytes(random_in[j], RAND_BYTES_x4));
+      ASSERT_TRUE(SHAKE128(random_in[j], RAND_BYTES_x4, digest[j],
+                                              RAND_OUT_BYTES_BLCKS * SHAKE128_BLOCKSIZE));
+    }
+
+    // Compute one batched x4 SHAKE128
+    ASSERT_TRUE(SHAKE128_Init_x4(&ctx));
+    ASSERT_TRUE(SHAKE128_Absorb_once_x4(&ctx, random_in[0], random_in[1], random_in[2], random_in[3],
+                                                                                          RAND_BYTES_x4));
+    ASSERT_TRUE(SHAKE128_Squeezeblocks_x4(digest_x4[0], digest_x4[1], digest_x4[2], digest_x4[3],
+                                                        &ctx, RAND_OUT_BYTES_BLCKS * SHAKE128_BLOCKSIZE));
```

Comment:
`Squeezeblocks` should take the number of blocks, not the number of bytes, and the multiplication with `SHAKE128_BLOCKSIZE` to get the number of bytes should be in the implementation.

Reply:
Thank you!